exporter/exporterhelper/internal/bounded_memory_queue.go

```diff
@@ -17,41 +16,21 @@ import (
 // where the queue is bounded and if it fills up due to slow consumers, the new items written by
 // the producer are dropped.
 type boundedMemoryQueue struct {
-	stopWG       sync.WaitGroup
-	stopped      *atomic.Bool
-	items        chan QueueRequest
-	numConsumers int
+	component.StartFunc
+	stopped *atomic.Bool
```

Comment:
Per our order of shutdown, we don't actually need this. Happy to review a PR after, and document that in our Queue implementation we guarantee that when Stop finishes we will not call Offer again (after you fix that re-enqueue logic).